dlr_import_amps.py

```diff
@@ -65,44 +65,48 @@ def setup_ksql(ksql_host: str, ksql_config: json):
                     print(f'Problem while trying to create {ksql_item["TYPE"].lower()}  \'{ksql_item["NAME"]}\'.')
                     return False
     else:
-        print('Error while gettings streams and tables from kSQL.')    
+        print('Error while gettings streams and tables from kSQL.')
         return False
     
-    print('kSQL setup has been validated.')
+    print('kSQL setup validated/created.')
     return True
 
-# Function call
-if setup_ksql(ksql_host, ksql_config):
-    print("kSQL setup validated/created.")
-else:
-    print("kSQL setup could not be validated/created.")
-
-# Main function
+# Trying kafka connection on first startup
 try:
     producer = KafkaProducer(bootstrap_servers=[IP],
                             value_serializer=lambda x: 
                             x.encode('utf-8'))
 except Exception:
-    print("Connection to kafka have failed. Please check enviroment variable 'IP' and recreate the container")
-    sys.exit(1)
+    print("Connection to kafka have failed. Please check enviroment variable 'IP' and if needed recreate the container")
 
+# Main function
 while True:
-    sleep(cycle)
+    start = time.time()
+    time.sleep(cycle)
+
+    # Starting with checking that the ksql streams and tables needed for the DLR algorithm exist
+    if not setup_ksql(ksql_host, ksql_config):
+        print('ksql failed validation')
+        continue
+    else:
+        print('ksql passed validation')
```

Comment:
Do you want to check the ksql state in every loop, or is it enough that it has passed validation once? 
You had a solution with the result of `setup_ksql` saved in a variable, which is great to check that validation has been passed once, eg
```suggestion
    if not ksql_validated:
        # Try setting up and validating again
        ksql_validated = setup_ksql(ksql_host, ksql_config)
        continue
```

Reply:
We have seen issues where ksql did not respond and did not know it was down itself therefor the ksql validation is inside the loop to make sure I catch if ksql is non responsive.